.github/workflows/push.yml

```diff
@@ -42,6 +42,53 @@ env:
   CC: ccache gcc
   CXX: ccache g++
 jobs:
+  CODE_FORMATTING_CHECK:
+    name: CODE_FORMATTING_CHECK
+    if: github.repository_owner == 'php' || github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    steps:
+      - name: git checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: apt
+        run: |
+          set -x
+          sudo apt-get update
+          sudo apt-get install \
+            git \
+            clang-format \
+            php-cli \
+            netcat-openbsd
+      - name: check code formatting
+        run: |
+          set -x
+          git fetch origin master
+          git diff origin/master > ../master.diff
+          if [ -f scripts/dev/format_pull_request.php ]; then
+            cp scripts/dev/format_pull_request.php ..
+          fi
+          git checkout master
+          if [ -f scripts/dev/format_pull_request.php ]; then
+            cp scripts/dev/format_pull_request.php ..
+          fi
+          if [ ! -f ../format_pull_request.php ]; then
+            echo "error: format_pull_request.php does not exist" 1>&2
+            exit 1
+          fi
+          git reset --hard origin/master
+          php ../format_pull_request.php ../master.diff > ../formatted.diff
+          if [ -s ../formatted.diff ]; then
+            cat ../formatted.diff
+            echo "code formatting check failed :("
+            patch_url=$(cat ../formatted.diff | nc termbin.com 9999)
+            echo "Please run"
+            echo "curl $patch_url | git apply -v"
```

Comment:
Think it would be better to just print it rather than save it to external service

Reply:
Currently do both. But how are authors supposed to easily apply the suggestions if we don't provide some easy copypasta command?

We could generate a 

>Please run
>echo base64(gzip(diff))  | base64 -d | grip -d | git apply -.

Along with the readable diff above 🤔

Reply:
I would have used `| patch -p1` and not `git apply -` because it doesn't make much sense to create a new commit.

Reply:
@Ayesh do you mean we can avoid doing
```
passthru('git -c user.email="clangformat@example.com" -c user.name=clangformat commit -a --message=clangformat --quiet', $ret);
``` 
By using patch instead of git apply?

Reply:
Yeah. My thinking is to to not commit anything by running the command we suggest. The user can then decide if they want to amend.

Reply:
If you mean automatic pushing to the PR, then please don't do that. :) This would be happening too often with our current format and would be probably quite annoying.

Reply:
There is no automatic pushing. The commit in format_pull_request.php is immediately reverted, it's part of setting up things for `clang-format-diff` to do it's thing, which was... tricky to get right